driver-mapping/src/main/java/com/datastax/driver/mapping/config/PropertyAccessStrategy.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.config;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+/**
+ * A strategy to determine how mapped properties are read and written.
+ */
+public interface PropertyAccessStrategy {
+
+    /**
+     * Returns whether or not the given {@link Field} should
+     * be used when accessing its underlying property.
+     *
+     * @return {@code true} if the given field should be considered, {@code false} otherwise.
+     */
+    boolean isFieldAccessAllowed();
+
+    /**
+     * Returns whether or not the given {@link PropertyDescriptor} should
+     * be used when accessing its underlying property.
+     *
+     * @return {@code true} if the given {@link PropertyDescriptor} should be considered, {@code false} otherwise.
+     */
+    boolean isGetterSetterAccessAllowed();
```

Comment:
It should be obvious that it is absurd to have both `isFieldAccessAllowed` and `isGetterSetterAccessAllowed` to false. But I'm mildly annoyed by the fact that the API allows it. That wasn't the case with the enum.